doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,696 @@
+            qml.RZ(x[0], wires=w)
+            qml.RX(x[1], wires=w)
+
+        return qml.expval(qml.X(0))
+
+>>> circuit([0.1, 0.2], range_of_wires=range(2))
+Array(0., dtype=float32)
+
+But, again, using JAX-compatible types wherever possible is recommended.
+
+Positional arguments
+~~~~~~~~~~~~~~~~~~~~
+
+Positional arguments in PennyLane are flexible in that their variable names can 
+instead be employed as keyword arguments (e.g., ``qml.RZ(0.1, wires=0)`` versus 
+``qml.RZ(phi=0.1, wires=0)``). However, to ensure differentiability and, in general,
+compatibility with program capture enabled, such arguments must be kept as positional, 
+regardless of whether they're provided as an acceptable JAX type. 
+
+For instance, consider this example with ``qml.RZ``:
+
+.. code-block:: python 
+
+    import jax.numpy as jnp
+
+    qml.capture.enable()
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit(angle):
+        qml.RX(phi=angle, wires=0)
+        return qml.expval(qml.Z(0))
+
+>>> angle = jnp.array(0.1)
+>>> circuit(angle)
+...
+UnexpectedTracerError: Encountered an unexpected tracer. A function transformed by JAX had a side effect, allowing for a reference to an intermediate value with type float32[] wrapped in a DynamicJaxprTracer to escape the scope of the transformation.
+...
+
+Even though the value for ``phi`` in ``qml.RZ`` is given as a valid JAX type, the 
+fact that it was provided as a keyword argument results in an error.
+
+But, when the angle is passed as a positional argument, the circuit executes as 
+expected:
+
+.. code-block:: python 
+
+    qml.capture.enable()
+
+    @qml.qnode(dev)
+    def circuit(angle):
+        qml.RX(angle, wires=0)
+        return qml.expval(qml.Z(0))
+
+>>> angle = jnp.array(0.1)
+>>> circuit(angle)
+Array(0.9950042, dtype=float32)
+
+Using program capture with Catalyst
+-----------------------------------
+
+To use the program capture feature with Catalyst, the ``qml.capture.enable()`` toggle
+is not required. Instead, when decorating a workflow with :func:`~.pennylane.qjit`, 
+add the ``experimental_capture=True`` flag:
+
+.. code-block:: python
+
+    dev = qml.device('lightning.qubit', wires=1)
+
+    @qml.qjit(experimental_capture=True)
+    @qml.qnode(dev)
+    def circuit():
+        qml.RX(0.1, wires=0)
+        return qml.state()
+
+>>> circuit()
+Array([0.99875026+0.j        , 0.        -0.04997917j], dtype=complex128)
+
+Transforms
+----------
+
+One of the core features of PennyLane is modularity, which has allowed users to 
+transform QNodes in a NumPy-like way and to create their own transforms with ease. 
+Your favourite transforms will still work with program capture enabled (including
+custom transforms), but decorating QNodes with just ``@transform_name`` **will not 
+work** and will give a vague error. Additionally, decorating QNodes with the experimental 
+:func:`~.pennylane.capture.expand_plxpr_transforms` decorator is required.
```

Comment:
This might not be a sharp bit! Waiting to see if this gets merged https://github.com/PennyLaneAI/pennylane/pull/7199